flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContextTest.java

```diff
@@ -182,6 +191,44 @@ public void testSnapshotAndRestore() throws Exception {
                 restoredTracker.assignmentsByCheckpointId());
     }
 
+    @Test
+    public void testCallableInterruptedDuringShutdownDoNotFailJob() throws InterruptedException {
+        AtomicReference<Throwable> expectedError = new AtomicReference<>(null);
+
+        ManuallyTriggeredScheduledExecutorService manualWorkerExecutor =
+                new ManuallyTriggeredScheduledExecutorService();
+        ManuallyTriggeredScheduledExecutorService manualCoordinatorExecutor =
+                new ManuallyTriggeredScheduledExecutorService();
+
+        SourceCoordinatorContext<MockSourceSplit> testingContext =
+                new SourceCoordinatorContext<>(
+                        manualCoordinatorExecutor,
+                        manualWorkerExecutor,
+                        new SourceCoordinatorProvider.CoordinatorExecutorThreadFactory(
+                                TEST_OPERATOR_ID.toHexString(), getClass().getClassLoader()),
+                        operatorCoordinatorContext,
+                        new MockSourceSplitSerializer(),
+                        splitSplitAssignmentTracker);
+
+        testingContext.callAsync(
+                () -> {
+                    throw new InterruptedException();
+                },
+                (ignored, e) -> {
+                    if (e != null) {
+                        expectedError.set(e);
+                        throw new RuntimeException(e);
+                    }
+                });
+
+        manualWorkerExecutor.triggerAll();
```

Comment:
It is needed here. `callAsync(...)` indirectly calls `manualWorkerExecutor.execute(...)`, which enqueues the runnable into an internal queue (in `ManuallyTriggeredScheduledExecutorService`) without actually executing the runnable. 

We need to call `manualWorkerExecutor.triggerAll()` to execute the runnable.

I verified that, if we remove `manualWorkerExecutor.triggerAll()`, the test would fail due to `assertTrue(expectedError.get() instanceof InterruptedException)` failure.